instrument.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package tally
+
+import (
+	"fmt"
+)
+
+const (
+	_resultType        = "result_type"
+	_resultTypeError   = "error"
+	_resultTypeSuccess = "success"
+	_timingFormat      = "%s.latency"
+)
+
+// NewInstrumentedCall returns an InstrumentedCall with the given name
+func NewInstrumentedCall(scope Scope, name string) InstrumentedCall {
+	return &instrumentedCall{
+		error:   scope.Tagged(map[string]string{_resultType: _resultTypeError}).Counter(name),
+		success: scope.Tagged(map[string]string{_resultType: _resultTypeSuccess}).Counter(name),
+		timing:  scope.Timer(fmt.Sprintf(_timingFormat, name)),
```

Comment:
I should've mentioned this earlier, we should replace `scope.Timer(fmt.Sprintf(fmt, name))` with `scope.SubScope(name).Timer(timingSuffix)` so that it uses the user configured `Separator` string in the `ScopeOptions` provided to the scope.

Reply:
Oh sweet. Didn't know about that method. Sounds good!